### PR TITLE
Pages and datatables

### PR DESCRIPTION
**Pull request by notks**

### CODE REVIEW

File: src/pages/recurringExpenses/index/RecurringExpenses.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+export function RecurringExpenses() {
+  const [t] = useTranslation();
+  const { dateFormat } = useCurrentCompanyDateFormats();
+  const pages = [
+    { name: t('recurring_expenses'), href: '/recurring_expenses' },
+  ];
+
+  const columns: DataTableColumns = [
+    {
+      //no status id enum
```
Comment (notks) [inline src/pages/recurringExpenses/index/RecurringExpenses.tsx:27]: I could not find status id enum in 
https://github.com/invoiceninja/invoiceninja/blob/v5-develop/app/Models/RecurringExpense.php
so i couldn't make status badge.

File: src/pages/recurringExpenses/index/RecurringExpenses.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      label: t('status'),
+    },
+    {
+      id: 'number',
+      label: t('number'),
+    },
+    { id: 'vendor_id', label: t('vendor') },
+
+    { id: 'client_id', label: t('client') },
+    //no frequency id enum
```
Comment (notks) [inline src/pages/recurringExpenses/index/RecurringExpenses.tsx:38]: I could not find status id enum in
https://github.com/invoiceninja/invoiceninja/blob/v5-develop/app/Models/RecurringExpense.php
so i couldn't make status badge.

File: src/pages/recurringInvoices/index/RecurringInvoices.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    },
+    {
+      id: 'frequency_id',
+      label: t('frequency'),
+    },
+    {
+      id: 'due_date_days',
+      label: t('due_date'),
+      format: (value) => {
+        //needs translation
```
Comment (notks) [inline src/pages/recurringInvoices/index/RecurringInvoices.tsx:67]: Does not have translation

File: src/pages/recurringInvoices/index/RecurringInvoices.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        if (value === 'terms') return t('use_payment_format');
```
Comment (notks) [inline src/pages/recurringInvoices/index/RecurringInvoices.tsx:68]: Does not have translation

File: src/pages/expenses/index/Expenses.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import { useTranslation } from 'react-i18next';
+
+export function Expenses() {
+  const [t] = useTranslation();
+  const { dateFormat } = useCurrentCompanyDateFormats();
+  const pages = [{ name: t('expenses'), href: '/expenses' }];
+  const columns: DataTableColumns = [
+    {
+      //no status field
+      id: 'status_id',
```
Comment (notks) [inline src/pages/expenses/index/Expenses.tsx:25]: no status id is received from api

File: src/pages/credits/index/Credits.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    {
+      id: 'date',
+      label: t('date'),
+      format: (value) => date(value, dateFormat),
+    },
+    {
+      id: 'paid_to_date',
+      label: t('remaining'),
+      //not sure if this check is ok since there is no "remaining" value
+      format: (value, resource) => {
```
Comment (notks) [inline src/pages/credits/index/Credits.tsx:45]: I don't know is this check meant to be done this way since no "remaining" field is found in api response.

**Review by beganovich — CHANGES_REQUESTED** [review]
Looks awesome, few changes needed:

1. Formatting client: It now should be possible using ClientResolver (pull latest `main`)

Recurring invoices table:
- If the number is empty, please show "Pending".
- "Due date" isn't formatted. These are values provided for constants. 
- "Frequency" isn't formatted.
- Clicking on client name should take us to single client page.

Payments table:
- Clicking on client name should take us to single client page.

Credits table:
- Clicking on client name should take us to single client page.

Projects table:
- Clicking on client name should take us to single client page.
- Apply `truncate` class on the public/private notes columns
- `budgeted_hours` typo (wrong reference on column)

Expenses table:
- If the invoice_id is set then it's invoiced, if should be invoiced is enabled it's pending otherwise it's logged (for showing status)
- Clicking on client name should take us to single client page.
- Apply `truncate` class on the public notes columns

Recurring invoices table:
- Missing status formatting
- Clicking on client name should take us to single client page.
- Apply `truncate` class on the public notes columns

File: src/pages/recurringExpenses/index/RecurringExpenses.tsx (hunk 1 above)
Comment (turbo124) [reply src/pages/recurringExpenses/index/RecurringExpenses.tsx]: Please use the same status id enum as Recurring Invoice for these:

    const STATUS_DRAFT = 1;
    const STATUS_ACTIVE = 2;
    const STATUS_PAUSED = 3;
    const STATUS_COMPLETED = 4;
    const STATUS_PENDING = -1;

File: src/pages/invoices/index/Invoices.tsx [hunk 7]
```diff
@@ -8,21 +8,64 @@
...
+    },
+    {
+      id: 'number',
+      label: t('number'),
+    },
+    {
+      id: 'client_id',
+      label: t('client'),
+      format: (value, resource) => (
+        <Link to={generatePath('/clients/:id', { id: resource.client.id })}>
```
Comment (beganovich) [inline src/pages/invoices/index/Invoices.tsx:39]: Wrong component. We should use <Link> from 'components/forms'. At the moment there's no visual identification it's clickable as it appears as any other text in the row.

**Review by beganovich — CHANGES_REQUESTED** [review]
Looks awesome, small change required :+1:

**Pull request merged**